c++/src/kj/compat/http.h

```diff
@@ -234,9 +234,20 @@ class HttpHeaderTable {
   kj::StringPtr idToString(HttpHeaderId id) const;
   // Get the canonical string name for the given ID.
 
+  bool isValid() const;
+  // Returns true if this HttpHeaderTable either was default constructed or its Builder has
+  // invoked `build()` and released it.
+
 private:
   kj::Vector<kj::StringPtr> namesById;
   kj::Own<IdsByNameMap> idsByName;
+
+  enum class BuildStatus {
+    UNSTARTED = 0,
+    BUILDING = 1,
+    FINISHED = 2,
+  };
+  BuildStatus buildStatus = BuildStatus::UNSTARTED;
```

Comment:
Could just be an `isReady` bool that defaults to true, but is set false by Builder's constructor, and then set true again in `build()`? The enum feels heavy to me, especially being int-sized rather than one byte. But I guess this is pretty nitpicky... I dunno.

Reply:
I think this should pad to be the same size in practice.